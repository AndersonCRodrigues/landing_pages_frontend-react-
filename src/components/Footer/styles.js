import styled, { css } from 'styled-components';

export const Container = styled.footer`
  ${({ theme }) => css`
    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
