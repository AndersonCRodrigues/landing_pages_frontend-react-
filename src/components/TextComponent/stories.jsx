import TextComponent from '.';

export default {
  title: 'TextComponent',
  component: TextComponent,
  args: {
    children: `
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
      Reprehenderit ipsum iusto voluptates eius.
      Quibusdam hic at, repellendus magnam soluta id facilis quia
      repudiandae voluptas ipsa. Minima impedit earum quam maiores?`,
  },
  argTypes: {
    children: { type: 'string' },
  },
};

export const Templates = (args) => {
  return (
    <div>
      <TextComponent {...args} />;
    </div>
  );
};
