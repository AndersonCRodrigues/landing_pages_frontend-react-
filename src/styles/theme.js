export const theme = {
  colors: {
    maninBg: 'red',
    secondaryBg: 'blue',
  },
};
