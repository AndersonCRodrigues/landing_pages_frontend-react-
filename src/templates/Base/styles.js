import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    padding-top: ${theme.spacings.huge};

    @media ${theme.media.lteMedium} {
      padding-top: 0;

    }
  `}
`;
