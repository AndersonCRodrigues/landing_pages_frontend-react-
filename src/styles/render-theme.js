import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from './theme';

export const renderTheme = (childen) => {
  return render(<ThemeProvider theme={theme}>{childen}</ThemeProvider>);
};
