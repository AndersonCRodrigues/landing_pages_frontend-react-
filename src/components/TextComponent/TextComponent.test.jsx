import { screen } from '@testing-library/react';
import TextComponent from '.';
import { renderTheme } from '../../styles/render-theme';

describe('<TextCompornent />', () => {
  it('should render TextComponent', () => {
    renderTheme(<TextComponent>texto</TextComponent>);

    expect(screen.getByText('texto')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<TextComponent>texto</TextComponent>);

    expect(container).toMatchSnapshot();
  });
});
