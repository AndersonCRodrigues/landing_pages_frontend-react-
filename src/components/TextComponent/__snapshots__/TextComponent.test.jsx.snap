// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextCompornent /> should match snapshot 1`] = `
.c0 {
  font-size: 2.4rem;
}

<div>
  <p
    class="c0"
  >
    texto
  </p>
</div>
`;
